### DISCUSSION TITLE
Performance tips for embedd systems

### DISCUSSION THREAD

**Discussion by MT00x** [bluenviron/mediamtx#1485] [Questions] [answered]
Hi, I'm using this great software on an ARMv5 successfully. 
In my scenario webcam stream video directly to rtsp server and client read the rtsp video(no re-encoding involved).
I would like to know if there are some suggestions to reduce cpu utilisation because cpu work seams a little bit high even if no client are reading the stream

Thanks

**Comment by aler9** [comment, marked as answer]
Hello, in order to reduce consumption on embedded hardware, you can:

* in case of CPU, disable the stream when there are no clients that are reading. For instance, if you're pulling the video of the webcam with RTSP, there's the option `sourceOnDemand: yes` that stops pulling the video when there are no connected clients for a certain amount of seconds:

   ```yml
   paths:
     cam:
       source: rtsp://my-cam
       sourceOnDemand: yes
   ```

* in case of RAM, if you're using HLS, you can use the new parameter `hlsDirectory` in order to save HLS segments on disk instead of on RAM:

   ```yml
   hlsDirectory: /home/pi/my-temp-dir
   ```

[a comment by an automated account is not shown]
